ballerina/listener_utils.bal

```diff
@@ -102,6 +106,176 @@ isolated function addDefaultDirectives(__Schema schema) {
+        files = fileInfoResult;
+    } else {
+        return createResponse(fileInfoResult, http:STATUS_BAD_REQUEST);
+    }
+    return forwardMultipartRequestToExecution(engine, context, files, payload);
+}
+
+isolated function handleFileField(mime:Entity bodyPart) returns FileUpload|error {
+    string encoding = bodyPart.getContentType();
+    string fileName = bodyPart.getContentDisposition().fileName;
+    mime:MediaType|mime:InvalidContentTypeError mediaType = mime:getMediaType(bodyPart.getContentType());
+    string|mime:HeaderNotFoundError contentEncoding = bodyPart.getHeader(CONTENT_ENCODING);
+    if contentEncoding is string {
+        encoding = contentEncoding;
+    }
+    if mediaType is mime:MediaType {
+        stream<byte[], io:Error?>|mime:ParserError byteStream = bodyPart.getByteStream();
+        if byteStream is stream<byte[], io:Error?> {
+            FileUpload file = {
+                fileName: fileName,
+                mimeType: mediaType.getBaseType(),
+                encoding: encoding,
+                byteStream: byteStream
+            };
+            return file;
+        }
+        return byteStream;
+    }
+    return mediaType;
+}
+
+isolated function getFileUploadValues(map<FileUpload> fileInfo, map<json> pathMap, map<json> variables) returns map<FileUpload|FileUpload[]>|string {
+    map<FileUpload> files = {}; //map of file path and file value
+    if pathMap.length() > 0 {
+        foreach var item in pathMap.entries() {
+            if item[1] is json[] {
+                json[] value = <json[]>item[1];
+                if value.length() > 0 {
+                    foreach json path in value {
+                        if path is string {
+                            if checkAndReplaceVariablePosition(variables, path) {
+                                string key = item[0];
+                                if fileInfo.hasKey(key) {
+                                    files[path] = <FileUpload> fileInfo[key];
+                                } else {
+                                    return "File content is missing in multipart request";
+                                }
+                            } else {
+                                return "Undefined variable found in multipart request `map`";
+                            }
+                        } else {
+                            return "Invalid type for multipart request field ‘map’ value";
+                        }
+                    }
+                } else {
+                    return "Missing multipart request field ‘map’ values";
+                }
+            } else {
+                return "Invalid type for multipart request field ‘map’";
+            }
+        }
+    } else {
+        return "Missing multipart request field ‘map’";
+    }
+    return createFileUploadInfoMap(files, variables);
+}
+
+isolated function checkAndReplaceVariablePosition(map<json> variables, string name) returns boolean {
```

Comment:
What is the meaning of the return value of this function? `checkAndReplaceVariablePosition` doesn't seem like a `boolean` returning function.